x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorFeatures.java

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.connector;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+public class ConnectorFeatures implements Writeable, ToXContentObject {
+
+    @Nullable
+    private final Boolean documentLevelSecurityEnabled;
```

Comment:
Would it be worth setting these to their default values instead of null if not specified? (Disclaimer: I have no strong feelings about this question at all).

Reply:
From my understanding the values are optional and tied to definitions in connectors framework. Can we leave it as it is, I will revisit this when implementing the endpoint to update those features?

Reply:
I think setting these booleans to false is perfectly fine. Making them nullable just adds complication without adding much value.

They're nullable in the protocol because these properties were added at later stages, so we had to take account documents that didn't have this property at all. But we have more control in this API so that's not really a concern, and we can just set them to `false` instead.